sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -778,8 +869,13 @@ def test_warm_start_oob_switch(Cls):
     est.fit(X, y)
 
     assert_array_equal(est.oob_improvement_[:100], np.zeros(100))
+    assert_array_equal(est.oob_scores_[:100], np.zeros(100))
+    assert est.oob_scores_[-1] == pytest.approx(est.oob_score_)
+
     # the last 10 are not zeros
     assert_array_equal(est.oob_improvement_[-10:] == 0.0, np.zeros(10, dtype=bool))
+    assert_array_equal(est.oob_scores_[-10:] == 0.0, np.zeros(10, dtype=bool))
```

Comment:
```suggestion
    assert (est.oob_improvement_[-10:] > 0.0).all()
    assert (est.oob_scores_[-10:] > 0.0).all()
```
nitpick/beyond this PR: I had to read these lines several times to figure out why the comment was true. In particular the "elements aren't zero" combined with "equals equals zero" made it hard. I think the above is easier to understand when you are trying to read the test (and convince yourself there are no bugs in the test). I think the improvement and scores should all be above zero? Or is this a weird thing where they could also be negative? If they can be negative we should leave it as is (at least I don't have a good suggestion for a rewrite)

Reply:
I see your point. The current check is quite confusing. 
Your suggestion is great. As the scores can be negative, we can just check if it is not equal to zero for the new values.